pkg/utils/split.go

```diff
@@ -11,3 +14,95 @@ var typesForPkgSplit = []ftypes.TargetType{ftypes.NodePkg, ftypes.Yarn, ftypes.N
 func IsPkgSplitRequired(targetType ftypes.TargetType) bool {
 	return lo.Contains(typesForPkgSplit, targetType)
 }
+
+// Func splits the given licenses (if any separators like AND/OR are present in the license),
+// validates those individual licenses if they are SPDX defined or not.
+// If any SPDX license is found, it filters those licenses and returns them.
+// else we just return the given input license string
+func FilterNGetLicenses(licenses []string) []string {
+	var foundSPDXLicense bool
+	var result []string
+	var tempLicenses []string
+
+	// Split and validate licenses
+	for _, license := range licenses {
+		// skip empty licenses
+		if len(license) == 0 {
+			continue
+		}
+		tempLicenses = append(tempLicenses, license)
+
+		for _, license := range splitLicense(license) {
+			ok, _ := spdxexp.ValidateLicenses([]string{license})
+			foundSPDXLicense = foundSPDXLicense || ok
+
+			if ok {
+				result = append(result, license)
+			}
+		}
+	}
+	licenses = tempLicenses
+
+	if !foundSPDXLicense {
+		return lo.Uniq(licenses)
+	}
+
+	// distinct license names will be returned
+	return lo.Uniq(result)
+}
+
+// Func parses the given license name.
+// If it's a single license, return it. If it's combination of licenses, split it based on the seperators and return it
+func splitLicense(license string) []string {
+	// common separators present in license name
+	separators := []string{" OR ", " AND "}
```

Comment:
Is there a specific standard followed? If not, it might be better to use regexes to extract substrings/parts

Reply:
I see that this OR and AND seperators are most commonly used when declaring licenses. Is regex processing required?

Reply:
There might be cases where there is an extra space etc. We'll miss those. If we have a regex then those cases can also be handled right?